src/env/emulated_syscalls/linux_syscalls.cpp

```diff
@@ -776,6 +813,177 @@ FunctionCallback::return_t sys_linux_exit(
+    const std::vector<Value>& args
+)
+{
+    return engine.process->egid;
+}
+
+// int getgroups(int size, gid_t list[]);
+FunctionCallback::return_t sys_linux_getgroups(
+    MaatEngine& engine,
+    const std::vector<Value>& args
+)
+{
+    // TODO: for now, no supplementary group IDs
+    engine.log.warning("Emulated getgroups(): not implemented yet");
+    return 0; // Success
+}
+
+// pid_t gettid(void);
+FunctionCallback::return_t sys_linux_gettid(
+    MaatEngine& engine,
+    const std::vector<Value>& args
+)
+{
+    // In a single-threaded process, the thread ID is equal to the process ID
+    return sys_linux_getpid(engine, args);
+}
+
+// long set_tid_address(int *tidptr);
+FunctionCallback::return_t sys_linux_set_tid_address(
+    MaatEngine& engine,
+    const std::vector<Value>& args
+)
+{
+    // TODO: for now, pretend everything went fine
+    engine.log.warning("Emulated set_tid_address(): faking success");
+    // Always return the caller's thread ID
+    return sys_linux_gettid(engine, args);
+}
+
+// long set_robust_list(struct robust_list_head *head, size_t len);
+FunctionCallback::return_t sys_linux_set_robust_list(
+    MaatEngine& engine,
+    const std::vector<Value>& args
+)
+{
+    // TODO: for now, pretend everything went fine
+    engine.log.warning("Emulated set_robust_list(): faking success");
+    return 0; // Success
+}
+
+// int rseq(struct rseq *rseq, uint32_t rseq_len, int flags, uint32_t sig);
+FunctionCallback::return_t sys_linux_rseq(
+    MaatEngine& engine,
+    const std::vector<Value>& args
+)
+{
+    // TODO: for now, pretend everything went fine
+    engine.log.warning("Emulated rseq(): faking success");
+    return 0; // Success
+}
+
+// int rt_sigaction(int signum, const struct sigaction *restrict act, struct sigaction *restrict oldact);
+FunctionCallback::return_t sys_linux_rt_sigaction(
+    MaatEngine& engine,
+    const std::vector<Value>& args
+)
+{
+    // TODO: for now, pretend everything went fine
+    engine.log.warning("Emulated rt_sigaction(): faking success");
+    return 0; // Success
+}
+
+// int syscall(SYS_rt_sigprocmask, int how, const kernel_sigset_t *set, kernel_sigset_t *oldset, size_t sigsetsize);
+FunctionCallback::return_t sys_linux_rt_sigprocmask(
+    MaatEngine& engine,
+    const std::vector<Value>& args
+)
+{
+    // TODO: for now, pretend everything went fine
+    engine.log.warning("Emulated rt_sigprocmask(): faking success");
+    return 0; // Success
+}
+
+// int prlimit(pid_t pid, int resource, const struct rlimit *new_limit, struct rlimit *old_limit);
+FunctionCallback::return_t sys_linux_prlimit64(
+    MaatEngine& engine,
+    const std::vector<Value>& args
+)
+{
+    addr_t old_limit = args[3].as_uint(*engine.vars);
+
+    // TODO: for now, no limits for whatever resource
+    if (old_limit) {
+        const auto RLIM_INFINITY = (1 << engine.arch->bits()) - 1;
+        const auto rlim_cur = RLIM_INFINITY;
+        const auto rlim_max = RLIM_INFINITY;
```

Comment:
```suggestion
        const auto INFINITY = (1 << engine.arch->bits()) - 1;
        const auto rlim_cur = INFINITY;
        const auto rlim_max = INFINITY;
```

Using `RLIM_INFINITY` directly causes compilation errors on macOS because the macro gets expanded by the compiler.